test/IntegrationTests/test_getallchats.py

```diff
@@ -0,0 +1,633 @@
+
+    def test_correct_customer_guid_and_wrong_chat_id(self):
+        """Test case 2: Valid customer_guid but invalid chat_id"""
+        logger.info("=== Starting Test Case 2: Correct customer_guid and wrong chat_id ===")
+
+        url = f"{self.BASE_URL}/getallchats"
+        data = {
+            "customer_guid": self.valid_customer_guid,  # Valid customer GUID
+            "chat_id": "invalid_chat_id"  # Invalid chat ID
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        # Check that the status code is 404
+        self.assertEqual(response.status_code, 404,
+                         "Expected status code 404 for valid customer_guid and invalid chat_id.")
+
+        # Verify that the response text contains the expected error message
+        self.assertIn("no chats found for this customer and chat id", response.text.lower(),
+                      "Expected error message not found in the response.")
+
+        logger.info("=== Test Case 2 Completed ===\n")
+
+    def test_wrong_customer_guid_and_correct_chat_id(self):
+        """Test case 3: Wrong customer_guid and correct chat_id"""
+        logger.info("=== Starting Test Case 3: Wrong customer_guid and correct chat_id ===")
+
+        url = f"{self.BASE_URL}/getallchats"
+        data = {
+            "customer_guid": "invalid_customer_guid",
+            "chat_id": self.valid_chat_id
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        # Check that the status code is 404
+        self.assertEqual(response.status_code, 404, "Expected status code 404 for invalid customer_guid.")
+
+        # Verify that the response text contains the expected error message
+        self.assertIn("no chats found for this customer and chat id", response.text.lower(),
+                      "Expected error message not found in the response.")
+
+        logger.info("=== Test Case 3 Completed ===\n")
+
+    def test_retrieval_with_valid_inputs(self):
+        """Test case 4: Valid customer_guid and chat_id"""
+        logger.info("=== Starting Test Case 4: Valid customer_guid and chat_id ===")
+
+        # Add multiple chat messages to the chat
+        chat_url = f"{self.BASE_URL}/chat"
+        messages_to_add = [
+            "Message 1: You will get the correct answer once AI is integrated.",
+            "Message 2: You will get the correct answer once AI is integrated.",
+            "Message 3: You will get the correct answer once AI is integrated.",
+
+        ]
+
+        for message in messages_to_add:
```

Comment:
You should get the chat_id from the first message and use it in further messages. I think you will get new chat ids for every message

Reply:
Done, sir. I've extracted the chat_id from the first message response and will use it for subsequent messages.